MetaMask/metamask-extension
# Snaps key access warning refactor

Fixes: #13297 

Explanation:  `SnapInstallWarning` component was re-written to use the `Popover` component instead of the deprecated `Modal` component. 

See the linked issue for an image of the previous UI and see below for the improved UI per [flask designs](https://www.figma.com/file/nBKUQx0btfQiqVf30UA6KD/Flask-MVP?node-id=336%3A6342)

<img src="https://user-images.githubusercontent.com/41640681/149455358-881030ef-0b57-4a51-9827-a0b6094a738e.png" width="300">


Manual testing steps:  
  - Run `yarn start --build-type flask`
  - To test a snap with the key access permissions, you can clone our fork of the filecoin snap at: [Filecoin snap](https://github.com/MetaMask/filsnap/tree/prerelease)
  - Checkout the `prerelease` branch and then run `yarn && yarn demo`
  - Follow the install flow for the filecoin snap from the locally hosted dapp and observe the changes when the snap install warning pops up
